src/main/java/com/Service/UserService.java

```diff
@@ -0,0 +1,53 @@
+package com.Service;
+
+import com.Model.ConfirmationToken;
+import com.Model.User;
+import com.Repository.ConfirmationTokenRepository;
+import com.Repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+import javax.servlet.http.HttpServletRequest;
+
+@Component
+public class UserService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private MessageSender messageSender;
+
+    @Autowired
+    private ConfirmationTokenRepository confirmationTokenRepository;
+
+    private final PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserService(PasswordEncoder passwordEncoder){
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public void save (User user, HttpServletRequest request){
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
```

Comment:
this makes me uneasy... it tells me for a moment the password is set in the clear.

Reply:
When user enter credentials in the website (or register through curl), those credentials are wrapped inside servlet request as a user object, this save method parse user object and encrypt data right after receiving the request.